jormungandr/src/topology/quarantine.rs

```diff
@@ -26,23 +26,30 @@ struct QuarantineRecord {
 pub struct Quarantine {
     quarantine_duration: StdDuration,
     quarantine_whitelist: HashSet<Address>,
-    quarantined_records: LruCache<PeerInfo, Instant>,
+    /// To avoid cycling down nodes back and and forth from quarantine (and as such prevent them
+    /// from being evicted from the lru cache), do not quarantine again nodes that were recently
+    /// lifted from quarantine.
+    ///
+    /// A peer is inserted in the grace list after being lifted from quarantine and is removed
+    /// from that list after we receive a new gossip about it.
+    quarantine_grace: LruCache<NodeId, ()>,
```

Comment:
We do not put any timeouts here and just allow these peers to be evicted from the cache when it is full, right?

Reply:
yep, either when it's full or we receive a new gossip about them, which means they are online again and we should not make special rules for them.